### ISSUE TITLE
Sim card removed error

### ISSUE THREAD

**Issue by MichaelXPO** [itsHardStyl3r/LG-G3-issue-tracker#10]
every couple of days my phone loses it's sim, saying there's no sim. i restart and it's recognized.
now with the latest builds of 16 it's worse, it loses the sim a few moments after booting.

**Comment by itsHardStyl3r** [comment]
Are you sure this is not hardware problem? Please provide more information and use labels.

**Comment by MichaelXPO** [comment]
yes, i am sure it's not a hardware problem, i flashed 17.1 and so far no sim loss. while a minute before on 16 the sim was not recognized or was lost in a matter of seconds from boot.

**Comment by itsHardStyl3r** [comment]
> what labels do you mean?
Labels on the right.
![image](https://user-images.githubusercontent.com/14148440/76698601-46b31580-66a5-11ea-8ec0-241f673e64a1.png)

Please keep this updated after I release March builds.

**Comment by MichaelXPO** [comment]
right now i'm on 17.1 and this only happened once since i flashed the rom. and that might have been a loose connection because it was after i took out the battery. but i will update you if there's anything new.

**Comment by MichaelXPO** [comment]
after a week of working with no hiccups on 17.1 it started again with the invalid sim error. it can't be a hardware problem because flashing a new version would not fix it for a week.

not sure what it could be.

**Comment by MichaelXPO** [comment]
Ok, i flashed the march version, it was quiet for a day, and then it started again.
i have more info that leads me to think it's software/driver related. 

now it's not only there's no sim or there's a sim and everything works, there's a middle ground where the phone recognizes the sim and shows full reception but when you try to call or send SMS (I'm not talking about internet data at all) the phone hangs on the dialer or just does not send the SMS.

i really don't want to flash the stock ROM to make sure it's the room, but I've been stranded without my phone being reachable in the last week way too many times. and somehow restarting it a few times fixes this.

if you need logs from my phone or anything to diagnose this further please let me know.

**Comment by itsHardStyl3r** [comment]
Please, give me some logs.

**Comment by MichaelXPO** [comment]
> Please, give me some logs.

which logs? and where are they located?

**Comment by itsHardStyl3r** [comment]
Take a full logcat.

**Comment by MichaelXPO** [comment]
ok, so once it happens, take a logcat and attach it here. got it.

**Comment by MichaelXPO** [comment]
[logcat.txt](https://github.com/itsHardStyl3r/LG-G3-issue-tracker/files/4388658/logcat.txt)
this is the logcat. after restart sim is recognized. I've also noticed when it reports reception but is not really connected the time on my phone was lagging by 10 minutes. could that help?

**Comment by itsHardStyl3r** [comment]
How did you determine it's not hardware issue? I couldn't find anything interesting in the logs and my G3 is running all good with SIM in.

**Comment by scepterus** [comment]
would a restart solve it if it's hardware? it looks like it's not transferring from lte to h+ correctly and loses the sim.

**Comment by Spasman** [comment]
Hey, I guess this git's dead but I had this problem too. I don't think it's the lineage build, I flashed back to stock and still had the problem. I did a bunch of research and it looks like the LG G3 has a hardware flaw with the SIM card slot, at least the variants where the slot is located next to the battery. Whether its from opening the phone to access the motherboard (like I did) or just from old age, the metal band that 'holds down' the SIM card seems to easily lose it's firmness and can shake the SIM card around and make it lose contact with the slot. Due to the nature of SIM cards needing to reboot the phone if 'removed', this can make any loss of contact with the slot a big problem.

When I took out the card for the first time in a few years, it was REALLY stuck in there, but afterwards it would slide in and out very easily. I suspected the sudden loosely fit SIM card was a symptom of this flaw I read about, and did what is known as the 'paper trick'. [This video](https://www.youtube.com/watch?v=zAWbmU8KxvU) helped me but I'll explain as well. I cut out a couple strips of printer paper, I made each strip a little long and gave them a pointed end. I then took out my SIM card and slipped in the paper strips and pulled up the pointed end with a micro screwdriver. Then a little firmly, I pulled up each end of the paper strips and while stuffing the SIM card underneath them, then trimmed off the excess paper.

A lot of advice says to only use 1-2 layers of paper, but it took me 4 layers to get it to connect consistently. I don't know if it helped, but after I put the phone back together and placed something heavy on top of it overnight, I don't know, I guess I was trying to squeeze out any remaining loose fit on the SIM card. I'm still on stock, but my signal connected on the first boot and has kept for a couple hours now, before it wasn't lasting more then 20 minutes. Using 4 layers of paper made it a really tight squeeze, and I could feel the contact points rubbing against the SIM card which made me nervous that I was damaging something, but that's when the SIM card actually started to connect again.

**Comment by Spasman** [comment]
Hey, I guess this git's dead but I had this problem too. I don't think it's the lineage build, I flashed back to stock and still had the problem. I did a bunch of research and it looks like the LG G3 has a hardware flaw with the SIM card slot, at least the variants where the slot is located next to the battery. Whether its from opening the phone to access the motherboard (like I did) or just from old age, the metal band that 'holds down' the SIM card seems to easily lose it's firmness and can shake the SIM card around and make it lose contact with the slot. Due to the nature of SIM cards needing to reboot the phone if 'removed', this can make any loss of contact with the slot a big problem.

When I took out the card for the first time in a few years, it was REALLY stuck in there, but afterwards it would slide in and out very easily. I suspected the sudden loosely fit SIM card was a symptom of this flaw I read about, and did what is known as the 'paper trick'. [This video](https://www.youtube.com/watch?v=zAWbmU8KxvU) helped me but I'll explain as well. I cut out a couple strips of printer paper, I made each strip a little long and gave them a pointed end. I then took out my SIM card and slipped in the paper strips and pulled up the pointed end by the lip with a micro screwdriver. Then a little firmly, I pulled up each end of the paper strips and while stuffing the SIM card underneath them, then trimmed off the excess paper.

A lot of advice says to only use 1-2 layers of paper, but it took me 4 layers to get it to connect consistently. I don't know if it helped, but after I put the phone back together and placed something heavy on top of it overnight, I don't know, I guess I was trying to squeeze out any remaining loose fit on the SIM card. I'm still on stock, but my signal connected on the first boot and has kept for a couple hours now, before it wasn't lasting more then 20 minutes. Using 4 layers of paper made it a really tight squeeze, and I could feel the contact points rubbing against the SIM card which made me nervous that I was damaging something, but that's when the SIM card actually started to connect again.

**Comment by Spasman** [comment]
Hey, I guess this git's dead but I had this problem too. I don't think it's the lineage build, I flashed back to stock and still had the problem. I did a bunch of research and it looks like the LG G3 has a hardware flaw with the SIM card slot, at least the variants where the slot is located next to the battery. Whether its from opening the phone to access the motherboard (like I did) or just from old age, the metal band that 'holds down' the SIM card seems to easily lose it's firmness and can shake the SIM card around and make it lose contact with the slot. Due to the nature of SIM cards needing to reboot the phone if 'removed', this can make any loss of contact with the slot a big problem.

When I took out the card for the first time in a few years, it was REALLY stuck in there, but afterwards it would slide in and out very easily. I suspected the sudden loosely fit SIM card was a symptom of this flaw I read about, and did what is known as the 'paper trick'. [This video](https://www.youtube.com/watch?v=zAWbmU8KxvU) helped me but I'll explain as well. I cut out a couple strips of printer/copier paper, I made each strip a little long and gave them a pointed end. I then took out my SIM card and slipped in the paper strips and pulled up the pointed end by the lip with a micro screwdriver. Then a little firmly, I pulled up each end of the paper strips and while stuffing the SIM card underneath them, then trimmed off the excess paper.

A lot of advice says to only use 1-2 layers of paper, but it took me 4 layers to get it to connect consistently. I don't know if it helped, but after I put the phone back together I placed something heavy on top of it overnight, I guess I was trying to squeeze out any remaining loose fit on the SIM card. I'm still on stock, but my signal connected on the first boot and has kept for a couple hours now, before it wasn't lasting more then 20 minutes. Using 4 layers of paper made it a really tight squeeze, and I could feel the contact points rubbing against the SIM card which made me nervous that I was damaging something, but that's when the SIM card actually started to connect again.
